crates/bevy_ecs/src/query/access.rs

```diff
@@ -2,24 +2,32 @@ use crate::storage::SparseSetIndex;
 use fixedbitset::FixedBitSet;
 use std::marker::PhantomData;
 
-/// `Access` keeps track of read and write accesses to values within a collection.
+/// Tracks read and write access to specific elements in a collection.
 ///
-/// This is used for ensuring systems are executed soundly.
+/// System initialization and execution uses these to ensure soundness.
+/// See the [`is_compatible`](Access::is_compatible) and [`get_conflicts`](Access::get_conflicts) functions.
 #[derive(Debug, Eq, PartialEq, Clone)]
 pub struct Access<T: SparseSetIndex> {
-    reads_all: bool,
-    /// A combined set of T read and write accesses.
+    /// The accessed elements.
     reads_and_writes: FixedBitSet,
+    /// The exclusively-accessed elements.
     writes: FixedBitSet,
+    /// Does this have access to all elements in the collection?
+    /// This field exists as a performance optimization for `&World` and `&mut World`.
+    reads_all: bool,
+    /// Does this have exclusive access to all elements in the collection?
+    /// This field exists as a performance optimization for `&mut World`.
+    writes_all: bool,
```

Comment:
So far these fields only exist for our internal use (for the `&World` and `&mut World` param)s.